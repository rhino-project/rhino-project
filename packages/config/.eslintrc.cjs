module.exports = {
  root: true,
  env: { browser: true, es2020: true },
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:deprecation/recommended',
    'plugin:react/recommended',
    'plugin:react-hooks/recommended',
    'plugin:prettier/recommended'
  ],
  plugins: ['react', 'react-hooks', 'react-refresh'],
  ignorePatterns: ['dist', '.eslintrc.cjs'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 'latest',
    sourceType: 'module',
    project: ['./tsconfig.json'],
    tsconfigRootDir: __dirname
  },
  overrides: [
    {
      files: ['src/**/*.{ts,tsx}'],
      extends: ['plugin:@typescript-eslint/recommended-type-checked']
    }
  ],
  rules: {
    'no-console': [
      'warn',
      {
        allow: ['assert', 'error', 'warn']
      }
    ],
    'no-restricted-globals': 'off',
    'no-var': 'error',
    'prefer-const': 'error',
    'react/jsx-key': 'off',
    'react/prop-types': 'off',
    'react/react-in-jsx-scope': 'off',
    'react-refresh/only-export-components': [
      'warn',
      {
        allowConstantExport: true,
        checkJS: true
      }
    ]
  },
  settings: {
    react: {
      version: 'detect'
    }
  }
};
