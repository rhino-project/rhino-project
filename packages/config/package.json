{
  "name": "@rhino-project/config",
  "version": "0.21.0-beta.26",
  "description": "",
  "author": "JP Rosevear",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/rhino-project/rhino-project.git",
    "directory": "packages/config"
  },
  "homepage": "https://rhino-project.org",
  "type": "module",
  "types": "dist/esm/index.d.ts",
  "main": "dist/cjs/index.cjs",
  "module": "dist/esm/index.js",
  "exports": {
    ".": {
      "import": {
        "types": "./dist/esm/index.d.ts",
        "default": "./dist/esm/index.js"
      },
      "require": {
        "types": "./dist/cjs/index.d.cts",
        "default": "./dist/cjs/index.cjs"
      }
    },
    "./assets": {
      "import": {
        "types": "./dist/esm/assets.d.ts",
        "default": "./dist/esm/assets.js"
      },
      "require": {
        "types": "./dist/cjs/assets.d.cts",
        "default": "./dist/cjs/assets.cjs"
      }
    },
    "./env": {
      "import": {
        "types": "./dist/esm/env.d.ts",
        "default": "./dist/esm/env.js"
      },
      "require": {
        "types": "./dist/cjs/env.d.cts",
        "default": "./dist/cjs/env.cjs"
      }
    },
    "./rhino-env": {
      "import": {
        "types": "./dist/esm/rhino-env.d.ts"
      },
      "require": {
        "types": "./dist/cjs/rhino-env.d.cts"
      }
    },
    "./package.json": "./package.json"
  },
  "sideEffects": false,
  "scripts": {
    "clean": "rimraf ./dist && rimraf ./coverage",
    "test:types": "tsc",
    "test:lib": "vitest",
    "test:lib:dev": "pnpm run test:lib --watch",
    "test:build": "publint --strict",
    "test:eslint": "eslint --ext .ts,.tsx --max-warnings 0 ./src ",
    "build": "vite build"
  },
  "keywords": [],
  "engines": {
    "node": ">=18"
  },
  "files": [
    "dist",
    "src"
  ],
  "dependencies": {
    "lodash-es": "4.17.21"
  },
  "devDependencies": {
    "@types/lodash-es": "^4.17.0",
    "@vitejs/plugin-react": "^4.2.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "rollup-plugin-copy": "^3.5.0"
  },
  "peerDependencies": {
    "react": ">=16",
    "react-dom": ">=16"
  }
}
