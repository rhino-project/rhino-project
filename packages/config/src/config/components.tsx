export type RhinoConfigComponentShort = React.ComponentType<unknown>;

export interface RhinoConfigComponentExpanded {
  component?: React.ComponentType<unknown>;
  props?: Record<string, unknown>;
}

export type RhinoConfigComponent =
  | RhinoConfigComponentShort
  | RhinoConfigComponentExpanded;

export type RhinoConfigGlobalComponentMap = {
  DisplayGroupArray: RhinoConfigComponent;
  DisplayGroupArrayReference: RhinoConfigComponent;
  DisplayGroupAttachment: RhinoConfigComponent;
  DisplayGroupAttachmentImage: RhinoConfigComponent;
  DisplayGroupAttachments: RhinoConfigComponent;
  DisplayGroupBoolean: RhinoConfigComponent;
  DisplayGroupCurrency: RhinoConfigComponent;
  DisplayGroupDate: RhinoConfigComponent;
  DisplayGroupDateTime: RhinoConfigComponent;
  DisplayGroupEnum: RhinoConfigComponent;
  DisplayGroupFloat: RhinoConfigComponent;
  DisplayGroupInteger: RhinoConfigComponent;
  DisplayGroupReference: RhinoConfigComponent;
  DisplayGroupString: RhinoConfigComponent;
  DisplayGroupText: RhinoConfigComponent;
  DisplayGroupTime: RhinoConfigComponent;
  DisplayGroupYear: RhinoConfigComponent;
  DisplayLabel: RhinoConfigComponent;
  DisplayLayoutFloating: RhinoConfigComponent;
  DisplayLayoutHorizontal: RhinoConfigComponent;
  DisplayLayoutVertical: RhinoConfigComponent;
  DisplayArray: RhinoConfigComponent;
  DisplayArrayReference: RhinoConfigComponent;
  DisplayBoolean: RhinoConfigComponent;
  DisplayCurrency: RhinoConfigComponent;
  DisplayDate: RhinoConfigComponent;
  DisplayDateTime: RhinoConfigComponent;
  DisplayEnum: RhinoConfigComponent;
  DisplayFloat: RhinoConfigComponent;
  DisplayImage: RhinoConfigComponent;
  DisplayInteger: RhinoConfigComponent;
  DisplayLink: RhinoConfigComponent;
  DisplayReference: RhinoConfigComponent;
  DisplayString: RhinoConfigComponent;
  DisplayStringControlled: RhinoConfigComponent;
  DisplayText: RhinoConfigComponent;
  DisplayTime: RhinoConfigComponent;
  FieldFeedback: RhinoConfigComponent;
  FieldGroupBoolean: RhinoConfigComponent;
  FieldGroupCountry: RhinoConfigComponent;
  FieldGroupCurrency: RhinoConfigComponent;
  FieldGroupDate: RhinoConfigComponent;
  FieldGroupDateTime: RhinoConfigComponent;
  FieldGroupFile: RhinoConfigComponent;
  FieldGroupFloat: RhinoConfigComponent;
  FieldGroupInteger: RhinoConfigComponent;
  FieldGroupPassword: RhinoConfigComponent;
  FieldGroupPhone: RhinoConfigComponent;
  FieldGroupSelectControlled: RhinoConfigComponent;
  FieldGroupString: RhinoConfigComponent;
  FieldGroupText: RhinoConfigComponent;
  FieldGroupTime: RhinoConfigComponent;
  FieldGroupYear: RhinoConfigComponent;
  FieldLabel: RhinoConfigComponent;
  FieldLayoutFloating: RhinoConfigComponent;
  FieldLayoutHorizontal: RhinoConfigComponent;
  FieldLayoutVertical: RhinoConfigComponent;
  FieldBoolean: RhinoConfigComponent;
  FieldBooleanIcon: RhinoConfigComponent;
  FieldBooleanIndeterminate: RhinoConfigComponent;
  FieldCountry: RhinoConfigComponent;
  FieldCurrency: RhinoConfigComponent;
  FieldDate: RhinoConfigComponent;
  FieldDateTime: RhinoConfigComponent;
  FieldFile: RhinoConfigComponent;
  FieldFloat: RhinoConfigComponent;
  FieldHidden: RhinoConfigComponent;
  FieldInput: RhinoConfigComponent;
  FieldInputControlled: RhinoConfigComponent;
  FieldInteger: RhinoConfigComponent;
  FieldPassword: RhinoConfigComponent;
  FieldPhone: RhinoConfigComponent;
  FieldSelect: RhinoConfigComponent;
  FieldSelectControlled: RhinoConfigComponent;
  FieldString: RhinoConfigComponent;
  FieldText: RhinoConfigComponent;
  FieldTime: RhinoConfigComponent;
  FieldYear: RhinoConfigComponent;
  FilterGroup: RhinoConfigComponent;
  FilterLabel: RhinoConfigComponent;
  FilterLayoutVertical: RhinoConfigComponent;
  CellBadge: RhinoConfigComponent;
  CellBoolean: RhinoConfigComponent;
  CellBooleanIcon: RhinoConfigComponent;
  CellCountry: RhinoConfigComponent;
  CellCurrency: RhinoConfigComponent;
  CellDate: RhinoConfigComponent;
  CellDateTime: RhinoConfigComponent;
  CellDateTimeDistance: RhinoConfigComponent;
  CellDateTimeRelative: RhinoConfigComponent;
  CellFloat: RhinoConfigComponent;
  CellImage: RhinoConfigComponent;
  CellInteger: RhinoConfigComponent;
  CellLink: RhinoConfigComponent;
  CellLinkEmail: RhinoConfigComponent;
  CellLinkTelephone: RhinoConfigComponent;
  CellReference: RhinoConfigComponent;
  CellString: RhinoConfigComponent;
  CellTime: RhinoConfigComponent;
};

export type RhinoConfigModelComponentMap = {
  ModelCreate: RhinoConfigComponent;
  ModelCreateActions: RhinoConfigComponent;
  ModelCreateForm: RhinoConfigComponent;
  ModelCreateHeader: RhinoConfigComponent;
  ModelCreateModal: RhinoConfigComponent;
  ModelCreateModalActions: RhinoConfigComponent;
  ModelDisplayLabel: RhinoConfigComponent;
  ModelEdit: RhinoConfigComponent;
  ModelEditActions: RhinoConfigComponent;
  ModelEditForm: RhinoConfigComponent;
  ModelEditHeader: RhinoConfigComponent;
  ModelEditModal: RhinoConfigComponent;
  ModelEditModalActions: RhinoConfigComponent;
  ModelFieldLabel: RhinoConfigComponent;
  ModelFilter: RhinoConfigComponent;
  ModelFilterGroup: RhinoConfigComponent;
  ModelFilterLabel: RhinoConfigComponent;
  ModelFilters: RhinoConfigComponent;
  ModelIndex: RhinoConfigComponent;
  ModelIndexActions: RhinoConfigComponent;
  ModelIndexHeader: RhinoConfigComponent;
  ModelIndexTable: RhinoConfigComponent;
  ModelPager: RhinoConfigComponent;
  ModelShow: RhinoConfigComponent;
  ModelShowActions: RhinoConfigComponent;
  ModelShowDescription: RhinoConfigComponent;
  ModelShowHeader: RhinoConfigComponent;
  ModelShowRelated: RhinoConfigComponent;
  AuthPage: RhinoConfigComponent;
};

export type RhinoConfigAttributeComponentMap = {
  ModelCellArray: RhinoConfigComponent;
  ModelCellArrayReference: RhinoConfigComponent;
  ModelCellAttachment: RhinoConfigComponent;
  ModelCellAttachmentDownload: RhinoConfigComponent;
  ModelCellAttachmentImage: RhinoConfigComponent;
  ModelCellAttachments: RhinoConfigComponent;
  ModelCellBoolean: RhinoConfigComponent;
  ModelCellBooleanIcon: RhinoConfigComponent;
  ModelCellCountry: RhinoConfigComponent;
  ModelCellCurrency: RhinoConfigComponent;
  ModelCellDate: RhinoConfigComponent;
  ModelCellDateTime: RhinoConfigComponent;
  ModelCellDateTimeDistance: RhinoConfigComponent;
  ModelCellEnum: RhinoConfigComponent;
  ModelCellFloat: RhinoConfigComponent;
  ModelCellInteger: RhinoConfigComponent;
  ModelCellLinkEmail: RhinoConfigComponent;
  ModelCellLinkTelephone: RhinoConfigComponent;
  ModelCellReference: RhinoConfigComponent;
  ModelCellString: RhinoConfigComponent;
  ModelCellTime: RhinoConfigComponent;
  ModelEditableCellBoolean: RhinoConfigComponent;
  ModelEditableCellReference: RhinoConfigComponent;
  ModelDisplayGroupArray: RhinoConfigComponent;
  ModelDisplayGroupArrayReference: RhinoConfigComponent;
  ModelDisplayGroupAttachment: RhinoConfigComponent;
  ModelDisplayGroupAttachmentImage: RhinoConfigComponent;
  ModelDisplayGroupAttachments: RhinoConfigComponent;
  ModelDisplayGroupBoolean: RhinoConfigComponent;
  ModelDisplayGroupCurrency: RhinoConfigComponent;
  ModelDisplayGroupDate: RhinoConfigComponent;
  ModelDisplayGroupDateTime: RhinoConfigComponent;
  ModelDisplayGroupEnum: RhinoConfigComponent;
  ModelDisplayGroupFloat: RhinoConfigComponent;
  ModelDisplayGroupInteger: RhinoConfigComponent;
  ModelDisplayGroupReference: RhinoConfigComponent;
  ModelDisplayGroupString: RhinoConfigComponent;
  ModelDisplayGroupText: RhinoConfigComponent;
  ModelDisplayGroupTime: RhinoConfigComponent;
  ModelFieldGroupArrayInteger: RhinoConfigComponent;
  ModelFieldGroupArrayString: RhinoConfigComponent;
  ModelFieldGroupBoolean: RhinoConfigComponent;
  ModelFieldGroupCountry: RhinoConfigComponent;
  ModelFieldGroupCurrency: RhinoConfigComponent;
  ModelFieldGroupDate: RhinoConfigComponent;
  ModelFieldGroupDateTime: RhinoConfigComponent;
  ModelFieldGroupEnum: RhinoConfigComponent;
  ModelFieldGroupFile: RhinoConfigComponent;
  ModelFieldGroupFloat: RhinoConfigComponent;
  ModelFieldGroupInteger: RhinoConfigComponent;
  ModelFieldGroupIntegerSelect: RhinoConfigComponent;
  ModelFieldGroupJoinSimple: RhinoConfigComponent;
  ModelFieldGroupNested: RhinoConfigComponent;
  ModelFieldGroupOwnerReference: RhinoConfigComponent;
  ModelFieldGroupPhone: RhinoConfigComponent;
  ModelFieldGroupReference: RhinoConfigComponent;
  ModelFieldGroupString: RhinoConfigComponent;
  ModelFieldGroupText: RhinoConfigComponent;
  ModelFieldGroupTime: RhinoConfigComponent;
  ModelFieldGroupYear: RhinoConfigComponent;
  ModelFieldNested: RhinoConfigComponent;
  ModelCell: RhinoConfigComponent;
  ModelDisplayGroup: RhinoConfigComponent;
  ModelFieldGroup: RhinoConfigComponent;
  ModelFooter: RhinoConfigComponent;
  ModelHeader: RhinoConfigComponent;
};
