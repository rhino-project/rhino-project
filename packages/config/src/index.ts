export * from './config';

// @ts-expect-error - We expect the file to exist in the project
export { default as CustomPrimaryNavigation } from 'components/app/CustomPrimaryNavigation';
// @ts-expect-error - We expect the file to exist in the project
export { default as CustomSecondaryNavigation } from 'components/app/CustomSecondaryNavigation';
