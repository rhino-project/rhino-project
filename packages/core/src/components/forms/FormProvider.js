import PropTypes from 'prop-types';
import { FormProvider as RHFFormProvider } from 'react-hook-form';
import { DevTool } from '@hookform/devtools';

const FormProvider = ({ children, ...props }) => {
  const { control } = props;

  return (
    <>
      {import.meta.env.DEV && (
        <DevTool
          control={control}
          placement={'top-left'}
          styles={{
            button: {
              width: '20px'
            }
          }}
        />
      )}
      <RHFFormProvider {...props}>{children}</RHFFormProvider>
    </>
  );
};

FormProvider.propTypes = {
  children: PropTypes.node
};

export default FormProvider;
