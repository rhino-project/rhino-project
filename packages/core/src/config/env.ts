export default {
  API_ROOT_PATH: import.meta.env.VITE_API_ROOT_PATH,
  DESIGN_SYSTEM_ENABLED:
    import.meta.env.DEV ||
    import.meta.env.VITE_DESIGN_SYSTEM_ENABLED === 'true',
  ROLLBAR_ACCESS_TOKEN: import.meta.env.VITE_ROLLBAR_ACCESS_TOKEN,
  ROLLBAR_ENV: import.meta.env.VITE_ROLLBAR_ENV || import.meta.env.MODE,
  ROLLBAR_ENABLED:
    import.meta.env.PROD || import.meta.env.VITE_ROLLBAR_ENABLED === 'true',
  STRIPE_PUBLISHABLE_KEY: import.meta.env.VITE_STRIPE_PUBLISHABLE_KEY
};
