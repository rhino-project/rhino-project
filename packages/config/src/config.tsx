import { useMemo } from 'react';
import env from './env';
import {
  RhinoConfigAttributeComponentMap,
  RhinoConfigGlobalComponentMap,
  RhinoConfigModelComponentMap
} from './components';
import merge from 'lodash-es/merge';

// We expect the user to create a file called rhino.config.js in the root of their project
// @ts-ignore
import rhinoConfig from 'rhino.config';

export type RhinoConfigGlobalComponent =
  | RhinoConfigGlobalComponentMap
  | RhinoConfigModelComponentMap
  | RhinoConfigAttributeComponentMap;

export type RhinoConfigModelComponent =
  | RhinoConfigModelComponentMap
  | RhinoConfigAttributeComponentMap;

export type RhinoConfigAttributeComponent = RhinoConfigAttributeComponentMap;

export interface RhinoConfig {
  version: number;
  appName: string;
  enableModelRoutes: boolean;

  // Config for env variables
  env: typeof env;

  // Config for global, model and attribute components
  components:
    | RhinoConfigGlobalComponent
    | Record<string, RhinoConfigModelComponent>
    | Record<string, Record<string, RhinoConfigAttributeComponent>>;
}

export const defaultConfig: RhinoConfig = {
  version: 1,
  appName: 'BoilerPlate',
  enableModelRoutes: true,
  env,
  components: {}
};

export const getRhinoConfig = (): RhinoConfig => {
  const mergedConfig = merge({}, defaultConfig, rhinoConfig);

  return mergedConfig;
};

export const useRhinoConfig = (): RhinoConfig =>
  useMemo(() => getRhinoConfig(), []);
