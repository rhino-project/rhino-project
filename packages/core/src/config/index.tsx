import { merge } from 'lodash';
import { useMemo } from 'react';
import rhinoConfig from 'rhino.config';
import env from 'rhino/config/env';
import {
  RhinoConfigAttributeComponentMap,
  RhinoConfigGlobalComponentMap,
  RhinoConfigModelComponentMap
} from './components';

export type RhinoConfigGlobalComponent =
  | RhinoConfigGlobalComponentMap
  | RhinoConfigModelComponentMap
  | RhinoConfigAttributeComponentMap;

export type RhinoConfigModelComponent =
  | RhinoConfigModelComponentMap
  | RhinoConfigAttributeComponentMap;

export type RhinoConfigAttributeComponent = RhinoConfigAttributeComponentMap;

export interface RhinoConfig {
  version: number;
  appName: string;
  enableModelRoutes: boolean;

  // Config for env variables
  env: typeof env;

  // Config for global, model and attribute components
  components:
    | RhinoConfigGlobalComponent
    | Record<string, RhinoConfigModelComponent>
    | Record<string, Record<string, RhinoConfigAttributeComponent>>;
}

export const defaultConfig: RhinoConfig = {
  version: 1,
  appName: 'BoilerPlate',
  enableModelRoutes: true,
  env,
  components: {}
};

export const getRhinoConfig = (): RhinoConfig => {
  const mergedConfig = merge({}, defaultConfig, rhinoConfig);

  return mergedConfig;
};

export const useRhinoConfig = (): RhinoConfig =>
  useMemo(() => getRhinoConfig(), []);
