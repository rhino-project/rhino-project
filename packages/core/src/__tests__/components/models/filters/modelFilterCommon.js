import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import { useForm } from 'react-hook-form';
import * as rhinoConfig from 'rhino.config';
import FormProvider from 'rhino/components/forms/FormProvider';

const components = [
  'ModelFilterBoolean',
  'ModelFilterDate',
  'ModelFilterDateTime',
  'ModelFilterEnum',
  'ModelFilterFloat',
  'ModelFilterInteger',
  'ModelFilterIntegerSelect',
  'ModelFilterFloat',
  // FIXME: These components need more complex setu
  // 'ModelFilterOwnerReference',
  // 'ModelFilterReference',
  'ModelFilterTime',
  'ModelFilterYear'
];

describe('Common model filter behaviour', () => {
  const modulePath = 'rhino/components/models/filters';
  const wrapper = ({ children }) => {
    const methods = useForm();
    return <FormProvider {...methods}>{children}</FormProvider>;
  };

  components.forEach((component) => {
    it(`should render ${component}`, async () => {
      const module = await import(`${modulePath}/${component}`);
      const Component = module.default;
      const { asFragment } = render(
        <Component
          model={{ properties: { id: { type: 'identifier' }, dummy: {} } }}
          path="dummy"
        />,
        {
          wrapper
        }
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it(`should pass down arbitrary props ${component}`, async () => {
      const module = await import(`${modulePath}/${component}`);
      const Component = module.default;
      const { asFragment } = render(
        <Component
          model={{ properties: { dummy: {} } }}
          path="dummy"
          data-foo="bar"
        />,
        {
          wrapper
        }
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
