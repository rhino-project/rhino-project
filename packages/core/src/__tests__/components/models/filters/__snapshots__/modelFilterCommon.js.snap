// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common model filter behaviour should pass down arbitrary props ModelFilterBoolean 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    class="form-check-input"
    data-foo="bar"
    id="dummy"
    name="dummy"
    path="dummy"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`Common model filter behaviour should pass down arbitrary props ModelFilterDate 1`] = `
<DocumentFragment>
  <div
    class="react-datepicker-wrapper d-block"
  >
    <div
      class="react-datepicker__input-container"
    >
      <input
        autocomplete="off"
        class="d-block form-control"
        id="dummy"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Common model filter behaviour should pass down arbitrary props ModelFilterDateTime 1`] = `
<DocumentFragment>
  <div
    class="react-datepicker-wrapper d-block"
  >
    <div
      class="react-datepicker__input-container"
    >
      <input
        autocomplete="off"
        class="d-block form-control"
        id="dummy"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Common model filter behaviour should pass down arbitrary props ModelFilterEnum 1`] = `
<DocumentFragment>
  <select
    aria-invalid="false"
    autocomplete="off"
    class="form-select"
    data-foo="bar"
    id="dummy"
    name="dummy"
    path="dummy"
  >
    <option
      disabled=""
      value="-1"
    >
      undefined...
    </option>
  </select>
</DocumentFragment>
`;

exports[`Common model filter behaviour should pass down arbitrary props ModelFilterFloat 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    data-foo="bar"
    id="dummy"
    name="dummy"
    path="dummy"
    type="number"
  />
</DocumentFragment>
`;

exports[`Common model filter behaviour should pass down arbitrary props ModelFilterFloat 2`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    data-foo="bar"
    id="dummy"
    name="dummy"
    path="dummy"
    type="number"
  />
</DocumentFragment>
`;

exports[`Common model filter behaviour should pass down arbitrary props ModelFilterInteger 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    data-foo="bar"
    id="dummy"
    name="dummy"
    path="dummy"
    type="number"
  />
</DocumentFragment>
`;

exports[`Common model filter behaviour should pass down arbitrary props ModelFilterIntegerSelect 1`] = `
<DocumentFragment>
  <select
    aria-invalid="false"
    autocomplete="off"
    class="form-select"
    data-foo="bar"
    id="dummy"
    name="dummy"
    path="dummy"
  />
</DocumentFragment>
`;

exports[`Common model filter behaviour should pass down arbitrary props ModelFilterTime 1`] = `
<DocumentFragment>
  <div
    class="react-datepicker-wrapper d-block"
  >
    <div
      class="react-datepicker__input-container"
    >
      <input
        class="d-block form-control"
        id="dummy"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Common model filter behaviour should pass down arbitrary props ModelFilterYear 1`] = `
<DocumentFragment>
  <div
    class="react-datepicker-wrapper d-block"
  >
    <div
      class="react-datepicker__input-container"
    >
      <input
        autocomplete="off"
        class="d-block form-control"
        id="dummy"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Common model filter behaviour should render ModelFilterBoolean 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    class="form-check-input"
    id="dummy"
    name="dummy"
    path="dummy"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`Common model filter behaviour should render ModelFilterDate 1`] = `
<DocumentFragment>
  <div
    class="react-datepicker-wrapper d-block"
  >
    <div
      class="react-datepicker__input-container"
    >
      <input
        autocomplete="off"
        class="d-block form-control"
        id="dummy"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Common model filter behaviour should render ModelFilterDateTime 1`] = `
<DocumentFragment>
  <div
    class="react-datepicker-wrapper d-block"
  >
    <div
      class="react-datepicker__input-container"
    >
      <input
        autocomplete="off"
        class="d-block form-control"
        id="dummy"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Common model filter behaviour should render ModelFilterEnum 1`] = `
<DocumentFragment>
  <select
    aria-invalid="false"
    autocomplete="off"
    class="form-select"
    id="dummy"
    name="dummy"
    path="dummy"
  >
    <option
      disabled=""
      value="-1"
    >
      undefined...
    </option>
  </select>
</DocumentFragment>
`;

exports[`Common model filter behaviour should render ModelFilterFloat 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    type="number"
  />
</DocumentFragment>
`;

exports[`Common model filter behaviour should render ModelFilterFloat 2`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    type="number"
  />
</DocumentFragment>
`;

exports[`Common model filter behaviour should render ModelFilterInteger 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    type="number"
  />
</DocumentFragment>
`;

exports[`Common model filter behaviour should render ModelFilterIntegerSelect 1`] = `
<DocumentFragment>
  <select
    aria-invalid="false"
    autocomplete="off"
    class="form-select"
    id="dummy"
    name="dummy"
    path="dummy"
  />
</DocumentFragment>
`;

exports[`Common model filter behaviour should render ModelFilterTime 1`] = `
<DocumentFragment>
  <div
    class="react-datepicker-wrapper d-block"
  >
    <div
      class="react-datepicker__input-container"
    >
      <input
        class="d-block form-control"
        id="dummy"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Common model filter behaviour should render ModelFilterYear 1`] = `
<DocumentFragment>
  <div
    class="react-datepicker-wrapper d-block"
  >
    <div
      class="react-datepicker__input-container"
    >
      <input
        autocomplete="off"
        class="d-block form-control"
        id="dummy"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
